### ISSUE TITLE
Dynamics overlap

### ISSUE THREAD

**Issue by JustTheEMailGuy** [musescore/MuseScore#20857]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

Dynamics overlap when multiple dyamics are chosen for one note. (or multiple if multiple are selected)

The correct behavior, as I understand it, should be that the dynamics change to the newest chosen.

### Steps to reproduce

1. Click on a note/select multiple notes over different staffs
2. Assign a dynamic to these notes
3. (Perhaps deselect)
4. If step 3 is needed then reselect
5. Assign dynamics to the note(s)
6. Notice how the dynamic markings will overlap and stack infinitly (if clicked an infinit amount of times)

### MuseScore Version

4.2.0.233521124

### Regression

I don't know

### Operating system

Windows 10

### Additional context

If you where to select the dynamic instead of the note, it behaves correctly and the dynamic is changed as intend.
The problem is that it's impossible to select just the dynamic over multiple staffs.
Clicking on every dynamic seperately is a little more tedious.

**Comment by MarcSabatella** [comment]
The replacing behavior is if you select the *original dynamic* when applying a new one.  If you select the *note*, then a second dynamic is added.  That seems to me to be by design, and in any case, it's desirable behavior because it's quite common in repeated passage to be able to notated multiple dynamics.

Of course, ideally they wouldn't just overlap.

**Comment by DaddyLudwig** [comment]
Yeah, I generally see something like p(f) or p-f for dynamic change upon repeat.

**Comment by JustTheEMailGuy** [comment]
Because they overlap, I didn't recognise it as a possible feature. If you place the same dynamic, it also stacks. As far as I remember, there is also no limit to how many can be stacked.
Well in that case, this is seemingly what should be changed.

Or

I would still like the ability to change a lot of dynamic markings at once.
So I'd have the dynamic be replaced when selecting the original dynamic and when the note is selected.

The probably much less used feature of adding a second marking, could then be assigned to the action of perhaps right clicking the original dynamic marking.

**Comment by MarcSabatella** [comment]
To be clear, you already can change a lot of dynamics at once.  Just select them (eg, click the first, Shift+click the last), then click the desired new dynamic in the palette.

I have no particular opinion on whether there should be alternative methods of doing this in addition, or whether they should take precedence over the way the palette normally works.  Just clearing up the confusion about how things currently work.

**Comment by JustTheEMailGuy** [comment]
That's helpful to know, thanks a lot!
I would argue that it's not intuative, but I'm also just one random user

**Comment by bkunda** [comment]
The options for alternative behaviour are: 

1. The dynamic gets replaced when the _note_ is selected. This, however, makes it harder to apply multiple dynamics on the same note (admittedly a pretty rare case).
2. Newly added dynamics get positioned immediately to the right of the existing ones (à-la adding a cresc._ or dim._ line to a dynamic).
3. Keep things as they are, bearing in mind it's still possible to replace multiple dynamics en mass without adding new ones (use the select > more option to select only dynamics at the same bar or beat position. It does the job, but it's admittedly a bit undiscoverable). 

I'm in favour of option 3 at the moment. No loss of functionality, and no additional dev time required. But perhaps option 2 should be considered as well. 

Thoughts @oktophonie?

**Comment by JustTheEMailGuy** [comment]
I'd like to add the option of the newly added dynamic appearing under the previous one!

To me that sounds like the neatest option.
It shows clearly that both dynamics belong to the same note and it looks less like an oversight.

But I understand that it takes effort.
If the dynamic above it is deleted, the one below should move up and take the place of the deleted one.
Otherwise it could become bothersome instead of helpful.

**Comment by FrancRos31** [comment]
Another instance: assign one dynamic to two or more adjacent notes.

https://github.com/user-attachments/assets/15795744-c925-4121-9d07-7e4de0df57b6

This is the latest master (february the 1st). #26132 already merged.

**Comment by JustTheEMailGuy** [comment]
I think with the new 4.5 update, this post has become redundant, right?
